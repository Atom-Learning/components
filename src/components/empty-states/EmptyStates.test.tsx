import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { EmptyStates } from '.'

const SimpleExample = () => (
  <EmptyStates size="lg">
    <EmptyStates.Title>No users found!</EmptyStates.Title>
    <EmptyStates.Body>
      You need to add some users before you can use this feature
    </EmptyStates.Body>
  </EmptyStates>
)

describe('EmptyStates component', () => {
  it('renders', async () => {
    const { container } = render(<EmptyStates />)

    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(<EmptyStates />)

    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders title and body correctly ', async () => {
    render(<SimpleExample />)
    expect(screen.getByText('No users found!')).toBeVisible()
    expect(
      screen.getByText(
        'You need to add some users before you can use this feature'
      )
    ).toBeVisible()
  })
})
