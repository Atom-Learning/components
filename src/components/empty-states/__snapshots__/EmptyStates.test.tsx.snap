// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyStates component renders 1`] = `
@media  {
  .c-dhzjXW {
    display: flex;
  }

  .c-kmANXN {
    flex-direction: column;
    align-items: center;
  }
}

@media  {
  .c-kmANXN-eSveXA-size-sm .c-eWgOGC {
    width: 84px;
    height: 48px;
    margin-bottom: var(--space-4);
  }

  .c-kmANXN-eSveXA-size-sm .c-ftwtsZ {
    font-size: var(--fontSizes-md);
    font-weight: 600;
    margin-bottom: var(--space-3);
  }

  .c-kmANXN-eSveXA-size-sm .c-cyRcZm {
    color: var(--colors-tonal400);
    font-size: var(--fontSizes-sm);
    font-weight: 400;
    margin-bottom: var(--space-4);
  }
}

<div>
  <div
    class="c-dhzjXW c-kmANXN c-kmANXN-eSveXA-size-sm"
  />
</div>
`;
